com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -504,6 +514,260 @@ internal void InvokeBehaviourOnGainedOwnership()
             }
         }
 
+        internal void InvokeBehaviourOnNetworkObjectParentChanged(NetworkObject parentNetworkObject)
+        {
+            for (int i = 0; i < ChildNetworkBehaviours.Count; i++)
+            {
+                ChildNetworkBehaviours[i].OnNetworkObjectParentChanged(parentNetworkObject);
+            }
+        }
+
+        private bool m_IsReparented; // Did initial parent (came from the scene hierarchy) change at runtime?
+        private ulong? m_LatestParent; // What is our last set parent NetworkObject's ID?
```

Comment:
as Luke raised in an internal conversation, when our parent NetworkObject is destroyed, we should clear our `m_LatestParent` here (otherwise, it's incorrect/broken).

Reply:
done.